src/streams/rtx_cache.rs

```diff
@@ -48,7 +48,7 @@ impl RtxCache {
     }
 
     pub fn get_cached_packet_smaller_than(&mut self, max_size: usize) -> Option<&mut RtpPacket> {
-        let quantized_size = max_size / RTX_CACHE_SIZE_QUANTIZER;
+        let quantized_size = max_size.min(DATAGRAM_MAX_PACKET_SIZE) / RTX_CACHE_SIZE_QUANTIZER;
```

Comment:
This should be `DATAGRAM_MTU_WARN - MAX_RTP_OVERHEAD`

Reply:
Are you sure ? It's like saying that the cache indexes up to 2000 but you're never ever allowed to request beyond 1200. How does it make sense ?

Reply:
str0m itself should always stay below DATAGRAM_MTU_WARN. However I can't control what people do with the RTP API, so they might push in something larger.

I don't want to clamp that larger thing, because then when we ask for something smaller, we might end up with the big one.

So the cache can take larger sizes, but we are not going to produce them for spurious padding.

I also think the `min` clamp should move back to where it was, because this is not an internal concern to RtxCache.

Reply:
So when they do push something larger, we do handle that bigger packet, but then we refuse to re use it for padding. Yet we do index it in write-only memory, and we rely on the caller to call the "rtx" cache api with a max_size that does not go out of bounds.

Alternatively, we could simply skip indexing the packet that exceeds the allowed padding size (instead of colliding them in the same slot) and keep the max rtx size a detail that is internal to the *rtx cache*

Reply:
Alright. Move the max size to a parameter to `new`. Throw away packages that are larger in `cache_sent_packet` and clamp to that size in `get_cached_packet_smaller_than`. Then set `DATAGRAM_MTU_WARN - MAX_RTP_OVERHEAD` on the outside when calling `new`.